types/state/stateTxn.go

```diff
@@ -0,0 +1,427 @@
+package state
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"github.com/AccumulateNetwork/accumulate/internal/logging"
+	"github.com/AccumulateNetwork/accumulate/smt/common"
+	"github.com/AccumulateNetwork/accumulate/smt/managed"
+	"github.com/AccumulateNetwork/accumulate/smt/pmt"
+	"github.com/AccumulateNetwork/accumulate/smt/storage"
+	"github.com/AccumulateNetwork/accumulate/smt/storage/database"
+	"github.com/AccumulateNetwork/accumulate/types"
+	"github.com/tendermint/tendermint/libs/log"
+	"sort"
+	"sync"
+	"time"
+)
+
+type DBTransactional struct {
```

Comment:
I would prefer this be named `DBTransaction`, since it represents a database transaction.